### PR TITLE
feature/shared file systems: show share details

### PR DESCRIPTION
**Pull request by mvalkon**
This pull request implements [`showing`][1] a [`share`][2]. 

Manila implementation related links:
https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/share/api.py#L1225

For #129  

[1]: http://developer.openstack.org/api-ref/shared-file-systems/?expanded=#show-share-details
[2]: http://docs.openstack.org/admin-guide/shared-file-systems-key-concepts.html#share

### CODE REVIEW

**Comment by mvalkon** [comment]
@jrperritt ping for review. This is based on #120

**Comment by mvalkon** [comment]
@jrperrit rebased on upstream, should be good to go

File: openstack/sharedfilesystems/v2/shares/testing/request_test.go [hunk 1]
```diff
@@ -31,3 +31,14 @@ func TestDelete(t *testing.T) {
...
+
+func TestGet(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+
+	MockGetResponse(t)
+
+	s, err := shares.Get(client.ServiceClient(), shareID).Extract()
+	th.AssertNoErr(t, err)
+	th.AssertEquals(t, s.ID, shareID)
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/shares/testing/request_test.go]: We want to make sure the JSON unmarshaling works for all the fields. Let's create the `Share` we expect and compare that with the actual one with `th.AssertDeepEquals`.

Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/testing/request_test.go]: Fair enough, should be fixed now.

**Comment by jrperritt** [comment]
+2

**Pull request merged**